js/ai/src/generate.ts

```diff
@@ -479,9 +479,39 @@ const isValidCandidate = (
   });
 };
 
-async function resolveModel(
-  model: ModelAction<any> | ModelReference<any> | string
-): Promise<ModelAction> {
+const DEFAULT_MODEL_GLOBAL_KEY = 'genkit_ai__defaultModel';
+
+interface GlobalModelRef {
+  model: ModelArgument<any>;
+  config?: any;
+}
+
+export function setGlobalDefaultModel<
+  CustomOptions extends z.ZodTypeAny = z.ZodTypeAny,
+>(model: ModelArgument<CustomOptions>, config?: z.infer<CustomOptions>) {
+  global[DEFAULT_MODEL_GLOBAL_KEY] = {
+    model,
+    config,
+  } as GlobalModelRef;
+}
```

Comment:
I'm curious - why not make this part of `configureGenkit`? Is it because models aren't a concept in core?

I ask because it could be kinda cool if `genkit init` set default models (e.g. Gemini Pro for Vertex/Google AI)

Reply:
Yeah. I can't strongly type it in the config (circular dep with AI package). We can still make genkit init set it this way.

Reply:
Can take a string model ref in the config....

Reply:
I think I'd probably prefer string model ref in config. Honestly I think we should consider merging `ai` and `core` packages in the future.

Reply:
Sure. Do you want config as well?

Reply:
Yes please, how about:

```ts
configureGenkit({
  defaults: {
    model: {ref: string | {name: string}, config: any}
  }
})
```

Reply:
How about:

```ts
interface {
  defaultModel?: {
    name: string | {name: string};
    config: Record<string,any>;
  }
}
```

Reply:
Sg

Reply:
Done.